onnx/onnx.proto

```diff
@@ -55,38 +56,28 @@ enum Version {
   IR_VERSION = 00000001;
 }
 
-// A named attribute containing either singular float, integer, string
-// and tensor values, or repeated float, integer, string and tensor values.
-// An AttributeProto MUST contain the name field, and *only one* of the
-// following content fields, effectively enforcing a C/C++ union equivalent.
-message AttributeProto {
-  // The name field MUST be present for this version of the IR.
-  optional string name = 1;           // namespace Attribute
-  optional float f = 2;               // float
-  optional int64 i = 3;               // int
-  optional bytes s = 4;               // UTF-8 string
-  optional TensorProto t = 5;         // tensor value
-  optional GraphProto g = 6;          // graph
-
-  repeated float floats = 7;          // list of floats 
-  repeated int64 ints = 8;            // list of ints
-  repeated bytes strings = 9;         // list of UTF-8 strings
-  repeated TensorProto tensors = 10;  // list of tensors
-  repeated GraphProto graphs = 11;    // list of graph
+// Defines information on value, including the name, the type, and
+// the shape of the value.
+message ValueInfoProto {
+  // This field MUST be present in this version of the IR.  
+  optional string name = 1;     // namespace Value
+  // This field MUST be present in this version of the IR.  
+  optional TypeProto type = 2;
+  optional TensorShapeProto shape = 3;
```

Comment:
Why not move TensorShapeProto into TensorTypeProto (as an optional field)?

Reply:
This will help address the issue of how the shape is described when we have, e.g., a sequence of records with one field of type tensor.